### PR TITLE
refactor prompt completer code

### PR DESCRIPTION
**Pull request by pazz**
This just splits the file completion.py, which was too large before, into several files.
We should to that with commands as well, but I figured I'd start with the completion code as it does not conflict (much) with open PRs..

### CODE REVIEW

**Comment by pazz** [comment]
The last commit addresses the codacy-bot issues; codacity issues can be ignored as they are just copies of the "fixed" ones from the removed `completion.py` file...

File: alot/completion/command.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+# This file is released under the GNU GPL, version 3 or a later revision.
+# For further details see the COPYING file
+
+import logging
+
+from alot import commands
+from alot.buffers import EnvelopeBuffer
+from alot.settings.const import settings
+from alot.utils.cached_property import cached_property
+from . completer import Completer
```
Comment (lucc) [inline alot/completion/command.py]: Why these spaces after the dots?

File: alot/completion/__init__.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from .completer import Completer
```
Comment (lucc) [inline alot/completion/__init__.py]: Why all these imports? They look unused to me.

Comment (pazz) [reply alot/completion/__init__.py]: yes they are. I just wanted to make sure nothing breaks by making these names available through `alot.completion` as before..

File: alot/commands/globals.py [hunk 3]
```diff
@@ -25,10 +25,10 @@
 from .. import helper
 from ..helper import split_commandstring
 from ..helper import mailto_to_envelope
-from ..completion import CommandLineCompleter
```
Comment (lucc) [inline alot/commands/globals.py]: Now I see what that file was about. I like it better the way it is now.

**Comment by pazz** [comment]
cheers for the quick response @lucc !

**Pull request merged**